cl/opinion_page/views.py

```diff
@@ -563,6 +583,28 @@ async def make_thumb_if_needed(
     return rd
 
 
+async def download_docket_entries_csv(
+    request: HttpRequest, docket_id: int
+) -> HttpResponse:
+    """Download csv file containing list of DocketEntry for specific Docket"""
+
+    docket, _ = await core_docket_data(request, docket_id)
+    de_list = await fetch_docket_entries(docket)
+    court_id = docket.court_id
+    case_name = docket.slug
+
+    date_str = datetime.datetime.now().strftime("%Y-%m-%d")
+    filename = f"{case_name}.{court_id}.{docket_id}_{date_str}.csv"
+
+    # TODO check if for large files we'll cache or send file by email
+    csv_content = await sync_to_async(generate_docket_entries_csv_data)(
+        de_list
+    )
```

Comment:
Our biggest issue here will be putting the file into memory before serving it. I wonder if we could stream it to the user as we generate it. I *think* all the pieces are there for this *but* if we do this *and* want to cache to disk, we'll need to stream to S3 and to the user. Now we're into the fun stuff...

Maybe later. :)